packages/auth/src/providers/cognito/utils/signInStore.ts

```diff
@@ -27,44 +30,117 @@ type Store<State, Action> = (reducer: Reducer<State, Action>) => {
 
 type Reducer<State, Action> = (state: State, action: Action) => State;
 
+const logger = new ConsoleLogger('Auth signInStore');
+
+// Minutes until stored session invalidates
+const EXPIRATION_MINUTES = 3;
+const MS_TO_EXPIRY = 1000 * 60 * EXPIRATION_MINUTES;
+const signInStateKeys = {
+	username: 'CognitoSignInState.username',
+	challengeName: 'CognitoSignInState.challengeName',
+	signInSession: 'CognitoSignInState.signInSession',
+	expiry: 'CognitoSignInState.expiry',
+};
+
 const signInReducer: Reducer<SignInState, SignInAction> = (state, action) => {
 	switch (action.type) {
 		case 'SET_SIGN_IN_SESSION':
 			return {
 				...state,
 				signInSession: action.value,
 			};
+
 		case 'SET_SIGN_IN_STATE':
+			persistSignInState(state);
```

Comment:
the `state` being passed in is not the latest `state` from the store. The `persistSignInState` should take the `action` as an argument instead.

```suggestion
persistSignInState(action.value);
```